### PR TITLE
[SDK-3244] Add Migration Guide

### PR DESCRIPTION
**Pull request by jimmyjames**
Adds a migration guide for updating to v4. @poovamraj would be good to have your eyes on it and ensure it captures everything and is clear, etc.

### CODE REVIEW

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+#### `MissingClaimException` added
+
+This class extends `InvalidClaimException` and represents that when validating a JWT, an expected claim is missing from the JWT.
+
+#### `HeaderParams` added
+
+This class contains constants representing common header parameter names.
+
+#### `StandardClaims` added
```
Comment (poovamraj) [inline MIGRATION_GUIDE.md:57]: Same as above, should be `RegisteredClaims`

File: MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+- When validating a JWT, if an expected claim is not present in the JWT, an `MissingClaimException` (subclass of `InvalidClaimException`) will now be thrown instead of an `InvalidClaimException`.
+- `withClaimPresence(String claimName)` now validates that the claim is present in the JWT, and a claim with a `null` value is considered present. Previously, a claim with a value of `null` would be considered as missing and fail the validation.
+- When validating a date/time claim value, the validation no longer checks for strict equality of the claim's value and the provided `Date` (or `Instant`). Instead, the expected `Date` or `Instant` will be compared to the claim's value only considering seconds (because JWT date/time claims are represented as seconds since the epoch).
+
+#### Claim changes
+
+- `com.auth0.jwt.interfaces.Claim#isNull()` now returns true only if the claim is present and its value is `null`. Previously, it returned true if the claim was present and its value was `null`, or if the claim was not present in the JWT. To check if the claim is present or not in the JWT, use `isMissing()`.
```
[the remark this answers is not in the document]
Comment (jimmyjames) [reply MIGRATION_GUIDE.md:41]: It is listed below in the "New classes or methods" section.

**Comment by jimmyjames** [comment]
New commit pushed to address review changes requested.

**Review by poovamraj — APPROVED** [review]
👍

**Pull request merged**